contracts/Polygon_SpokePool.sol

```diff
@@ -164,10 +171,51 @@ contract Polygon_SpokePool is IFxMessageProcessor, SpokePool {
         _wrap();
     }
 
+    /**
+     * @notice These functions can send an L2 to L1 message so we are extra cautious about preventing a griefing vector
+     * whereby someone batches this call with a bunch of other calls and produces a very large L2 burn transaction.
+     * This might make the L2 -> L1 message fail due to exceeding the L1 calldata limit.
+     */
+
+    function executeUSSRelayerRefundLeaf(
+        uint32 rootBundleId,
+        USSRelayerRefundLeaf memory relayerRefundLeaf,
+        bytes32[] memory proof
+    ) public override {
+        // AddressLibUpgradeable.isContract isn't a sufficient check because it checks the contract code size of
+        // msg.sender which is 0 if called from a constructor function on msg.sender. This is why we check if
+        // msg.sender is equal to tx.origin which is fine as long as Polygon supports the tx.origin opcode.
+        // solhint-disable-next-line avoid-tx-origin
+        if (relayerRefundLeaf.amountToReturn > 0 && msg.sender != tx.origin) revert NotEOA();
+        super.executeUSSRelayerRefundLeaf(rootBundleId, relayerRefundLeaf, proof);
```

Comment:
Why don't we need to do the _revertIfFunctionCalledAtomically/_setFunctionLock logic in this function?

Reply:
shoot, we do. I forgot to add